frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/tags/DisplayDownloadReport.java

```diff
@@ -8,41 +10,37 @@
  */
 package org.eclipse.sw360.portal.tags;
 
-import com.google.common.base.Function;
-import com.google.common.collect.FluentIterable;
-import com.google.common.collect.ImmutableMap;
-import com.google.common.collect.Maps;
 import org.eclipse.sw360.datahandler.thrift.attachments.Attachment;
 import org.eclipse.sw360.datahandler.thrift.attachments.AttachmentType;
-import org.apache.thrift.TBase;
 
 import javax.servlet.jsp.JspException;
 import javax.servlet.jsp.JspTagException;
 import javax.servlet.jsp.JspWriter;
+import javax.servlet.jsp.jstl.core.LoopTagStatus;
 import javax.servlet.jsp.jstl.core.LoopTagSupport;
+
 import java.io.IOException;
+import java.util.Collections;
 import java.util.Iterator;
-import java.util.Map;
 import java.util.Set;
 
-import static com.google.common.base.Predicates.equalTo;
-import static com.google.common.collect.Maps.transformValues;
-import static org.eclipse.sw360.datahandler.common.CommonUtils.afterFunction;
 import static org.eclipse.sw360.datahandler.common.CommonUtils.nullToEmptySet;
-import static org.eclipse.sw360.datahandler.thrift.ThriftUtils.extractField;
-import static org.eclipse.sw360.datahandler.thrift.attachments.Attachment._Fields.ATTACHMENT_CONTENT_ID;
-import static org.eclipse.sw360.datahandler.thrift.attachments.Attachment._Fields.ATTACHMENT_TYPE;
-import static org.eclipse.sw360.datahandler.thrift.attachments.Attachment._Fields.FILENAME;
 
 /**
- * This displays a Download Report attachment if available
- *
- * @author daniele.fognini@tngtech.com
+ * Displays a list of download links for attachments of the specified type.
  */
 public class DisplayDownloadReport extends LoopTagSupport {
 
-    private FluentIterable<Attachment> attachments;
-    private Iterator<Map.Entry<String, String>> attachmentIterator;
+    protected AttachmentType attachmentType;
+    protected Set<Attachment> unfilteredAttachments = Collections.emptySet();
+    protected Iterator<Attachment> attachmentIterator = unfilteredAttachments.iterator();
+
+    @Override
+    protected void prepare() throws JspTagException {
+        this.attachmentIterator = unfilteredAttachments.stream().filter((attachment) -> {
+            return attachmentType == null || attachment.getAttachmentType() == attachmentType;
```

Comment:
From the naming of the class I would say that the class is able to list all attachments regardless of their types. I've seen filter as an additional possibility not as an requirement. That was the reason why I accepted null.
If we always want to filter I would suggest to rename the class to something like 'DisplayAttachmentTypeReport' and mark the attribute as required.

I think both solutions are ok, what do you think?